src/test/java/org/mockito/CustomMockMakerTest.java

```diff
@@ -0,0 +1,189 @@
+/*
+ * Copyright (c) 2022 Mockito contributors
+ * This program is made available under the terms of the MIT License.
+ */
+package org.mockito;
+
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertThrows;
+import static org.mockito.Mockito.withSettings;
+
+import org.junit.Test;
+import org.mockito.exceptions.base.MockitoException;
+import org.mockito.exceptions.verification.SmartNullPointerException;
+import org.mockito.internal.creation.bytebuddy.ByteBuddyMockMaker;
+import org.mockito.internal.creation.bytebuddy.InlineByteBuddyMockMaker;
+import org.mockito.internal.creation.bytebuddy.SubclassByteBuddyMockMaker;
```

Comment:
> Unfortunately, we can't expose these classes in Mockito.java, since we don't know which mockmaker we should load.

Maybe we could use static methods to make the class objects available without loading or exposing the class?

```java
public final class MockMakers {
  private MockMakers() {} // Cannot be instantiated

  public Class<? extends MockMaker> inline() { return InlineByteBuddyMockMaker.class; };
  public Class<? extends MockMaker> subclass() { return SubclassByteBuddyMockMaker.class; };
  // ...
}
```

In contrast to constants, I think this should only load the classes for which you call the method. I am unsure why it is important to not load the classes, though.

Reply:
I wonder if we should instead use the String aliases for the mockmakers. E.g. you can set an enum here, of which the string value will correspond with the alias defined in https://github.com/mockito/mockito/blob/de63c24352d035dff77e9ef8e17481d9d85e87d9/src/main/java/org/mockito/internal/configuration/plugins/DefaultMockitoPlugins.java#L22

Then we would need to introduce an alias for the subclass mockmaker as well, such as `mockito-subclass-maker`?

Reply:
I guess strings or providing two separate options are our only possibility. At least I cannot think of anything else. I am not sure where you want me to use an enum. I would just use a class with string constants instead.

Reply:
Yeah agreed. The easiest solution here is to allow users to specify the String fully qualified name of the mockmaker. Since we don't want to expose the internal mockmaker locations as public API, for Mockito itself we would need to support the alias for the `mock-maker-inline` and `mock-maker-subclass`.

And disregard the comment regarding enums. That wouldn't work for community mockmakers, which would be a major benefit for them. If we lock the possible values to just the enum, that wouldn't work for them. Good point 👍

Reply:
@JojOatXGME Shall we go for the String solution including the default alias for `mock-maker-inline` and a new alias `mock-maker-subclass`?

Reply:
At least, I don't have a good idea for an alternative. I adjusted the implementation. However, I duplicated the code for the aliases for the time being. I was not sure how to share the code between `MockUtil.getMockMaker` and `PluginInitializer.loadImpl` without making bigger changes.